rfcs/accessibility-checklist.md

```diff
@@ -0,0 +1,52 @@
+# Blade Component Accessibility Checklist <!-- omit in toc -->
+
+This document will contain information and guidance on how to ensure accessibility of components,
+And how we can incorporate it in the component development life cycle.
+
+**5 Minute Guide for ensuring accessibility in blade components:**
+
+
+https://user-images.githubusercontent.com/35374649/175493627-4c5f29e9-0625-44fe-a5a8-03df7700f886.mp4
+
+
+## Component Decisions
+
+While writing `decision.md` file for component make sure to include `Accessibility` section and link to the appropriate [WAI ARIA component pattern](https://www.w3.org/WAI/ARIA/apg/patterns/) 
+
+Example: 
+
+Let's say you are building a `Checkbox` component. 
+
+- Step 1: Write the core logic for checkbox behaviour 
+- Step 2: Find the appropriate [WAI-ARIA Pattern for checkbox](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/)
+- Step 3: Accordion to the pattern implement [proper keyboard accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/#keyboard-interaction-5), like in checkbox we have to make sure to pressing <kbd>Space</kbd> should toggle the state. 
+- Step 4: Each pattern will have [an example section](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/#examples-1) which you can check for implementation and detailed information about [Accessibility Features](https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html#accessibilityfeatures), [Keyboard Support](https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html#kbd_label) and what various [aria attributes](https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html#rps_label) the component needs to be accessible. 
+  
+Following these patterns & examples we can make our components accessible.  
+
+## Implementations
+
+To implement the proper aria attributes we can make use of our `makeAccessible` function which we already [discussed in the RFC](https://github.com/razorpay/blade/blob/master/rfcs/2022-04-09-accessibility.md#platform-specific-implementation--5). 
```

Comment:
can we create a link to the example you created in recipes for `makeAccessible` usage?